File: Backend/Backend.Domain/Entity/Trail.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Backend.Domain.Entity;

public class Trail
{
   [Key]
   public int?  Id { get; init; }
   public string? Name { get; set; }
   public List<string>? Images { get; set; }
   public string? Description { get; set; }
   public string? TrailType { get; set; }
   public string? Location { get; set; }
   public double? Rating { get; set; }
   public string? Difficulty { get; set; }
  
 public int? MapId;
public Map? Maps { get; set; }

    [JsonIgnore]
public UserActivites? UserActivites { get; set; }
    [JsonIgnore]
public List<Comments>? Comments { get; set; }

}

File: frontend/lib/Features/Admin/Domain/entity/trail.dart
import 'dart:convert';

class Trail {
  String name;
  List<String> images;
  String description;
  String trailType;
  double length;
  double? elevation;
  String difficulty;
  String? location;

  Trail(
      {required this.name,
      required this.images,
      required this.description,
      required this.trailType,
      required this.length,
      this.elevation,
      required this.location,
      required this.difficulty});

  Map<String, dynamic> toJson() {
    return {
      "name": name,
      "images": images,
      "description": description,
      "trailType": trailType,
      "length": length,
      "elevation": elevation ?? 0,
      "difficulty": difficulty,
      "location": location
    };
  }

  factory Trail.fromJson(Map<String, dynamic> json) {
    return Trail(
        name: json['name'],
        images: json['images'].cast<String>(),
        description: json['description'],
        trailType: json['trailType'],
        length: json['length'],
        elevation: json['elevation'],
        difficulty: json['difficulty'],
        location: json['location']);
  }
}
